### DISCUSSION TITLE
ensure workers are on seperate nodes?

### DISCUSSION THREAD

**Discussion by cbc02009** [cloudnative-pg/cloudnative-pg#770] [Q&A]
All the workers seem to be running on just one of my nodes currently. Using the helm release, what annotations would I set to make sure the workers are all running on separate nodes? Or does it not really matter what node they're running on?

**Comment by gbartolini** [comment]
By default, the operator tries to schedule pods in different nodes, if available. See: https://cloudnative-pg.io/documentation/current/scheduling/#pod-affinity-and-anti-affinity

In particular, by default we set `topologyKey: kubernetes.io/hostname`. This means that, if you have multiple nodes where the required resources can be scheduled, the operator out of the box will try and place them in separate nodes. Hence my question: how many nodes are available?

Regarding your last question about deciding which node they are running, I think that in some cases this is actually what we want and, to be honest, one of the real advantages of using Kubernetes for our virtual infrastructure management (via taints, node selectors, and so on). The page I linked should contain all the information you need to answer your questions above. If not, please let us know.

**Reply by cbc02009** [reply]
there are 3 nodes available. One master and two workers. I think you may be on to something, since adding
```yaml
  affinity:
    enablePodAntiAffinity: true #default value
    topologyKey: kubernetes.io/hostname #defaul value
    podAntiAffinityType: required
```
to my cluster config causes the second instance to be stuck at pending with 
```
  Warning  FailedScheduling  3m45s  default-scheduler  0/3 nodes are available: 3 pod has unbound immediate PersistentVolumeClaims. preemption: 0/3 nodes are available: 3 Preemption is not helpful for scheduling.
  Warning  FailedScheduling  3m43s  default-scheduler  0/3 nodes are available: 1 node(s) didn't match pod anti-affinity rules, 1 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 2 node(s) had volume node affinity conflict. preemption: 0/3 nodes are available: 1 No preemption victims found for incoming pod, 2 Preemption is not helpful for scheduling.
```
Thanks so much for pointing me in the right direction, I don't know enough about k8s yet to know about antiAffinity. I'll do some more digging.

While I have you here (so I don't need to make another discussion) any idea when the helm chart will be updated with the pod security policy fixes so that it will be compatible with k8s 1.25? I'd love to be able to update my cluster, but no hurry of course. I know it's probably not high on the todo list.
